TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/FullTrafficReplayerTest.java

```diff
@@ -41,6 +43,7 @@
 // to the test server, a shutdown will stop those work threads without letting them flush through all of their work
 // (since that could take a very long time) and some of the work might have been followed by resource releases.
 @WrapWithNettyLeakDetection(disableLeakChecks = true)
+@Execution(ExecutionMode.SAME_THREAD)
```

Comment:
Can you set this for the whole class?

Reply:
This can apply on either whole class or individual methods with different behavior as described best in [baeldung blog](https://www.baeldung.com/junit-5-parallel-tests#:~:text=(same_thread%2C%20concurrent)%20%E2%80%93%20tests,but%20each%20class%20run%20separately).

I've since moved this to use ResourceLock as that's more accurate in behavior to what we're needing